src/abstractdataframe/abstractdataframe.jl

```diff
@@ -606,7 +606,7 @@ julia> df
 ```
 """
 Base.filter!(f, df::AbstractDataFrame) =
-    deleterows!(df, findall(!f, eachrow(df)))
+    deleterows!(df, findall(collect(!f(r)::Bool for r in eachrow(df))))
```

Comment:
Why is this change necessary? Because `eachrow` isn't indexable and `findall` requires an indexable collection?

Reply:
Yes. That's kind of annoying, IIRC in the PR it was mentioned that a wrapper could be introduced in Base to use linear indices in such situations. But here it's probably faster to allocate a vector anyway (https://github.com/JuliaLang/julia/pull/25879), and we do that in `deleterows`.